odoo/modules/loading.py

```diff
@@ -224,9 +224,14 @@ def load_module_graph(
         loaded_modules.append(package.name)
         if needs_update:
             models_updated |= set(model_names)
-            models_to_check -= set(model_names)
             registry._setup_models__(env.cr)
             registry.init_models(env.cr, model_names, {'module': package.name}, new_install)
+            # check custom models again, discard normal models from checks
+            for model_name in model_names:
+                if env.registry[model_name]._custom:
+                    models_to_check.add(model_name)
+                else:
+                    models_to_check.discard(model_name)
```

Comment:
`models_to_check` now are models which are simply loaded again after updated + custom_models if and module is updated? Am I right? Maybe we should defined it somewhere since `models_to_check` is quite vague

since `custom_models` are models not updated during the updating modules.
sementically I expect 🤔 
`models_update |= set(non_custom_model_names`
or maybe we should rename `models_update`?

Reply:
Custom models are updated during updating of modules (that's the root cause of the issue here). If you define a model "base" like `web` does, it will update custom models. These are not defined in that module so they should be checked afterwards. Normal models have been checked, so they can be discarded.

`models_to_check` is used in quite a few places and serves as a container that is passed into multiple functions. The variable keeps the same meaning: all models that need to be loaded again to update their definition in the DB.
I wanted to isolate just this change and we can refactor afterwards when the need arises.